disnake/welcome_screen.py

```diff
@@ -0,0 +1,223 @@
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from .partial_emoji import PartialEmoji, _EmojiTag
+from .utils import MISSING, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from .emoji import Emoji
+    from .guild import Guild
+    from .invite import PartialInviteGuild
+    from .state import ConnectionState
+    from .types.welcome_screen import (
+        WelcomeScreen as WelcomeScreenPayload,
+        WelcomeScreenChannel as WelcomeScreenChannelPayload,
+    )
+__all__ = ("WelcomeScreen", "WelcomeScreenChannel")
+
+
+class WelcomeScreenChannel:
+    """Represents a Discord welcome screen channel.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The ID of the guild channel this welcome screen channel represents.
+    description: :class:`str`
+        The description of this channel in the official UI.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji associated with this channel's welcome message, if any.
+    """
+
+    __slots__ = (
+        "id",
+        "description",
+        "emoji",
+    )
+
+    def __init__(
+        self,
+        *,
+        id: int,
+        description: str,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = None,
+    ):
+        self.id: int = id
+        self.description: str = description
+        self.emoji: Optional[PartialEmoji] = None
+        if emoji is None:
+            self.emoji = None
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, PartialEmoji):
+            self.emoji = emoji
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji._to_partial()
+        else:
+            raise TypeError("emoji must be None, a str, PartialEmoji, or Emoji instance.")
+
+    def __repr__(self):
+        return f"<WelcomeScreenChannel id={self.id!r} emoji={self.emoji!r} description={self.description!r}>"
+
+    @classmethod
+    def _from_data(
+        cls, *, data: WelcomeScreenChannelPayload, state: ConnectionState
+    ) -> WelcomeScreenChannel:
+        emoji_id = _get_as_snowflake(data, "emoji_id")
+        emoji_name = data.get("emoji_name") or None
+        emoji = None
+        if emoji_name:
+            emoji = PartialEmoji.with_state(state, name=emoji_name, id=emoji_id)
+
+        return cls(id=int(data["channel_id"]), description=data["description"], emoji=emoji)
+
+    def to_dict(self) -> WelcomeScreenChannelPayload:
+
+        result = {
+            "channel_id": self.id,
+            "description": self.description,
+        }
+        if self.emoji is not None:
+            if self.emoji.id:
+                result["emoji_id"] = self.emoji.id
+            result["emoji_name"] = self.emoji.name
+
+        return result  # type: ignore
```

Comment:
```suggestion
        result: WelcomeScreenChannelPayload = {}  # type: ignore
        result["channel_id"] = self.id
        result["description"] = self.description
        if self.emoji is not None:
            if self.emoji.id:
                result["emoji_id"] = self.emoji.id
            result["emoji_name"] = self.emoji.name

        return result
```
improves typing a bit